src/utils/safeBatchGetItem.js

```diff
@@ -0,0 +1,111 @@
+const _ = require('lodash');
+
+// DO NOT MODIFY, this value is specified in dynamodb docs
+const maxKeys = 100;
+
+const mergeIntoBatchKeysObject = (keys, tableName, batchKeysObject) => {
+  const sumOfKeys = _.sumBy(Object.values(batchKeysObject), (obj) => obj.length);
+
+  const spaceInRequest = maxKeys - sumOfKeys;
+
+  // If the new keys fit in the current object then just return it with the new entry
+  if (keys.length <= spaceInRequest) {
+    return [{ ...batchKeysObject, [tableName]: keys }];
+  }
+
+  const firstKeys = keys.slice(0, spaceInRequest);
+  const secondKeys = keys.slice(spaceInRequest);
+
+  // Put as many keys as we can fit in old object
+  const updatedBatchObj = {
+    ...batchKeysObject,
+    [tableName]: firstKeys,
```

Comment:
Here you may be overwriting the items from the same table that we got in the previous call (I may be wrong).
```
const tableName = 'tableName';
const batchKeysObject = {[tableName]: [1,2,3]}
const firstKeys = [4,5,6]
const updatedBatchObj = { ...batchKeysObject, [tableName]: firstKeys}
...
{ tableName: [ 4, 5, 6 ] }
```

I would try to go for more readibility at the cost of some potential extra call to `batchGetItem` (in real life, are we going to use `safeBatchGetItem` passing different tables in the same request?):
```
for every table
  create chunks of 100 at most
    push chunk into batchGetKeys
...
```

Reply:
We won't overwrite because each iteration is over a different tableName

Reply:
Oh, I see. Thanks!